sdks/dotnet/.github/workflows/github-actions.yml

```diff
@@ -11,3 +22,68 @@ jobs:
 
       - name: Ensure no changes in Generated Code
         run: ./bin/check-clean-git-status
+
+  publish:
+    runs-on: ubuntu-latest
+    if: github.repository == 'hellosign/dropbox-sign-dotnet'
+    needs: [ build ]
+    env:
+      BUILD_CONFIG: 'Release'
+      SOLUTION: 'Dropbox.Sign'
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v3
+
+      - name: Setup .NET
+        uses: actions/setup-dotnet@v3
+        with:
+          dotnet-version: 6.0.x
+
+      - name: Restore dependencies
+        run: dotnet restore ${SOLUTION}.sln
+
+      - name: Build
+        run: dotnet build ${SOLUTION}.sln --configuration ${BUILD_CONFIG} --no-restore
+
+      - name: Run tests
+        run: dotnet test /p:Configuration=${BUILD_CONFIG} --no-restore --no-build --verbosity normal
+
+      - name: Run pack
+        run: dotnet pack --configuration ${BUILD_CONFIG} ${SOLUTION}.sln
+
+      # This runs on merging to "main" branch
+      # Builds and publishes package to nuget.org
+      - name: Publish to nuget
+        if: github.ref == 'refs/heads/main' && github.event_name != 'pull_request'
+        run: dotnet nuget push src/${SOLUTION}/bin/Release/*.nupkg -k ${NUGET_API_KEY} -s https://api.nuget.org/v3/index.json
+        env:
+          NUGET_API_KEY: ${{ secrets.NUGET_API_KEY }}
+
+  # This job runs on merging to "main" branch
+  # Creates a new tag using the value in the VERSION file
+  cut-tag:
+    runs-on: ubuntu-latest
+    if: >-
+      github.repository == 'hellosign/dropbox-sign-dotnet'
+      && github.ref == 'refs/heads/main'
+      && github.event_name != 'pull_request'
+    needs: [ publish ]
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v3
+
+      - name: Retrieve version
+        run: echo "PACKAGE_VERSION=$(cat VERSION)" >> $GITHUB_ENV
+
+      - name: Create tag
+        uses: actions/github-script@v6
+        env:
+          PACKAGE_VERSION: ${{ env.PACKAGE_VERSION }}
+        with:
+          script: |
+            github.rest.git.createRef({
```

Comment:
If I understand this [doc](https://octokit.github.io/rest.js/v19#git-create-tag) correctly, this step creates the tag/reference in git, but not in github. We may want to create a tag object as well. Additionally, this is the [way](https://octokit.github.io/rest.js/v19#repos-create-release) to create a release I think (`actions/github-script` is a wrapper of `octokit`, the official typescript API client). Feel free address this in the next PR.

Reply:
I ran a test in the Ruby repo and saw the tag was properly created in github. I'm assuming they do some further magic themselves 🤷🏼